### PR TITLE
feat(CC): add oauth login to samples

### PR DESCRIPTION
**Pull request by bhabalan**
# COMPLETES #[SPARK-583528](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-583528)

## This pull request addresses

Need to do all testing with an integration that has minimal scopes instead of copying from the dev portal or agent desktop

## by making the following changes

Added oauth login to cc samples

Vidcast: https://app.vidcast.io/share/86fafaad-afd3-4218-b885-78aa90ca34a3

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios were tested

- Login via OAuth
- Initialize webex
- Register CC
- Station login via extension
- List teams
- Set agent status to idle / available
- List buddy agents
- Accept a task on device
- Observe that hold/resume works

### Scopes used:

``` JS
ccMandatoryScopes = [
    "cjp:config_read",
    "cjp:config_write",
    "cjp:config",
    "cjp:user",
  ];

const webRTCCallingScopes = [
    "spark:webrtc_calling",
    "spark:calls_read",
    "spark:calls_write",
    "spark:xsi"
  ];
  
const additionalScopes = [
    "spark:kms", // to avoid token downscope to only spark:kms error on SDK init
  ];
  ```
  
  The integration was created using the credentials listed in https://confluence-eng-gpk2.cisco.com/conf/x/C86lIw
  More details about the discovery are listed in the same confluence

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request

- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [x] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

## Summary by CodeRabbit

- **New Features**
	- Introduced OAuth authentication capabilities alongside existing access token methods in the contact center application.
	- Added a sample interface for the Webex JavaScript SDK, including sections for authentication, registration, and agent management.

- **Bug Fixes**
	- Enhanced error handling in the buddy agent retrieval process.

- **Documentation**
	- Updated HTML structure for improved user navigation and interaction with the Webex SDK functionalities.

### CODE REVIEW

**Comment by bhabalan** [comment]
@coderabbitai review

**Review by sreenara — CHANGES_REQUESTED** [review]
Please list the manual test scenarios in the PR description as well more details about the integration and its current scopes.

**Comment by bhabalan** [comment]
@coderabbit review

File: docs/samples/contact-center/app.js [hunk 1]
```diff
@@ -70,6 +77,65 @@ function toggleDisplay(elementId, status) {
...
+        .concat(webRTCCallingScopes)
+        .concat(additionalScopes))
+      ).join(' ');
+
+  webex = window.webex = Webex.init({
+    config: generateWebexConfig({
+      credentials: {
+        client_id: 'C70599433db154842e919ad9e18273d835945ff198251c82204b236b157b3a213',
+        client_secret: '575ba9f5034f8a28dfef2770870c50bfc6e0b2b749f14e6a14845a1a47622f87',
+        redirect_uri: redirectUri,
```
[the remark this answers is not in the document]
Comment (bhabalan) [reply docs/samples/contact-center/app.js]: We are okay with this as it's an isolated sandbox account for testing

File: docs/samples/contact-center/app.js [hunk 2]
```diff
@@ -83,6 +149,13 @@ function generateWebexConfig({credentials}) {
   };
 }
 
+// SPARK-499535
```
Comment (mkesavan13) [inline docs/samples/contact-center/app.js]: Why do we have this spark ID here?

File: docs/samples/contact-center/app.js [hunk 3]
```diff
@@ -146,9 +219,9 @@ function register() {
         }
 
         const idleCodesList = agentProfile.idleCodes;
-        
+
```
Comment (mkesavan13) [inline docs/samples/contact-center/app.js]: Why do we need these empty lines?

**Review by mkesavan13 — APPROVED** [review]
Just a couple of nitpicks. Please ensure to have addressed/responded to all the code rabbit reviews. Approving

File: docs/samples/contact-center/app.js (hunk 2 above)
Comment (bhabalan) [reply docs/samples/contact-center/app.js]: Removed

File: docs/samples/contact-center/app.js (hunk 3 above)
Comment (bhabalan) [reply docs/samples/contact-center/app.js]: Resolved

**Review by sreenara — APPROVED** [review]

**Pull request merged**